mesheryctl/internal/cli/root/perf/profile.go

```diff
@@ -0,0 +1,195 @@
+
+	"github.com/ghodss/yaml"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/constants"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/models"
+	"github.com/pkg/errors"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	pageSize = 25
+	expand   bool
+)
+
+type profileStruct struct {
+	Name           string
+	ID             *uuid.UUID
+	TotalResults   int
+	Endpoints      []string
+	QPS            int
+	Duration       string
+	Loadgenerators []string
+	ServiceMesh    string
+	LastRun        *sql.Time
+}
+
+var profileCmd = &cobra.Command{
+	Use:   "profile [profile-name]",
+	Short: "List performance profiles",
+	Long:  `List all the available performance profiles`,
+	Args:  cobra.MinimumNArgs(0),
+	Example: `
+// List performance profiles (maximum 25 profiles)	
+mesheryctl perf profile 
+
+// List performance profiles with search (maximum 25 profiles)
+mesheryctl perf profile test profile 2 
+
+// View performance profiles with more information
+mesheryctl perf profile --expand
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		// used for searching performance profile
+		var searchString string
+		// setting up for error formatting
+		cmdUsed = "profile"
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return ErrMesheryConfig(err)
+		}
+
+		profileURL := mctlCfg.GetBaseMesheryURL() + "/api/user/performance/profiles"
+
+		// set default tokenpath for command.
+		if tokenPath == "" {
+			tokenPath = constants.GetCurrentAuthToken()
+		}
+
+		if len(args) > 0 {
+			// Merge args to get profile-name
+			searchString = strings.Join(args, "%20")
+		}
+
+		data, expandedData, body, err := fetchPerformanceProfiles(profileURL, searchString)
+		if err != nil {
+			return err
+		}
+
+		if len(data) == 0 {
+			log.Info("No Performance Profiles to display")
+			return nil
+		}
+
+		if outputFormatFlag != "" {
+			if outputFormatFlag == "yaml" {
+				body, _ = yaml.JSONToYAML(body)
+			} else if outputFormatFlag != "json" {
+				return ErrInvalidOutputChoice()
+			}
+			log.Info(string(body))
+		} else if !expand {
+			utils.PrintToTable([]string{"Name", "ID", "RESULTS", "Load-Generator", "Last-Run"}, data)
+		} else {
+			for _, a := range expandedData {
+				fmt.Printf("Name: %v\n", a.Name)
+				fmt.Printf("ID: %s\n", a.ID.String())
+				fmt.Printf("Total Results: %d\n", a.TotalResults)
+				fmt.Printf("Endpoint: %v\n", a.Endpoints[0])
+				fmt.Printf("Load Generators: %v\n", a.Loadgenerators[0])
+				fmt.Printf("Test run duration: %v\n", a.Duration)
+				fmt.Printf("QPS: %d\n", a.QPS)
+				fmt.Printf("Service Mesh: %v\n", a.ServiceMesh)
+				if a.LastRun != nil {
+					fmt.Printf("Last Run: %v\n", a.LastRun.Time.Format("2006-01-02 15:04:05"))
+				} else {
+					fmt.Printf("Last Run: %v\n", "nil")
+				}
+				fmt.Println("#####################")
```

Comment:
Can we have a new line to separate different profiles/results instead of this?

Is this consistent with other similar commands?

Reply:
This is consistent across similar commands. It's probably better separator for eyes as sometimes it printes more than 10 profiles at the same time.

Reply:
@piyushsingariya I see places where `---------------------` is used as well.

Reply:
I see `----------------` used in some places as well.